server/src/com/thoughtworks/go/config/update/TemplateConfigCommand.java

```diff
@@ -46,7 +46,7 @@ public TemplateConfigCommand(PipelineTemplateConfig templateConfig, LocalizedOpe
     public boolean isValid(CruiseConfig preprocessedConfig) {
         TemplatesConfig templatesConfig = preprocessedConfig.getTemplates();
         preprocessedTemplateConfig = findAddedTemplate(preprocessedConfig);
-        preprocessedTemplateConfig.validate(ConfigSaveValidationContext.forChain(preprocessedConfig));
+        preprocessedTemplateConfig.validate(ConfigSaveValidationContext.forChain(templatesConfig));
```

Comment:
The preprocessed config is not being validated elsewhere later. We skip it because we only want to validate a specific entity(template) and related entities (eg. templates). 
> Since there is a much smaller part of the config being sent here, are you sure there is no other part of the config which was being validated here that isn't any more?

The template config validation only deals with its own object and nothing else as far as I know.